src/Microsoft.VisualStudio.Threading.Shared/AsyncBarrier.cs

```diff
@@ -25,24 +25,22 @@ public class AsyncBarrier
         private readonly int participantCount;
 
         /// <summary>
-        /// The number of participants that have not yet signaled the barrier.
-        /// </summary>
-        private int remainingParticipants;
-
-        /// <summary>
         /// The set of participants who have reached the barrier, with their awaiters that can resume those participants.
         /// </summary>
-        private ConcurrentStack<TaskCompletionSource<bool>> waiters;
+        private Stack<TaskCompletionSource<EmptyStruct>> waiters;
```

Comment:
True, we'd save the allocation of the Stack itself, but I'm not too concerned about the fixed cost of AsyncBarrier creation. It's the runtime allocation count from usage that might be concerning at some point -- if we ever get a user for this class outside of unit tests.